extensions/amp-facebook/0.1/amp-facebook.js

```diff
@@ -84,6 +85,13 @@ class AmpFacebook extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    const embedAs = this.element.getAttribute('data-embed-as');
+    userAssert(
+      !embedAs || ['post', 'video', 'comment'].indexOf(embedAs) !== -1,
```

Comment:
This assertion was moved into `amp-facebook` (from the `amp-facebook` case inside `3p/facebook.js` so that the newly supported values (`comments`, `page`, `like`) cannot be instantiated from this component in addition to their AMP components (`amp-facebook-comments`, `amp-facebook-page`, `amp-facebook-like`).

Reply:
While we _can_ eventually widen `data-embed-as` values to cover all other `amp-facebook-*` component types, we should not immediately as `amp-facebook` and `amp-facebook-comments` resize via `forceChangeHeight` whereas the other two resize via `attemptChangeHeight` -- consolidating all facebook component types to be embed types of `amp-facebook` would mean potentially introducing layout shift on the `page` and `like` embed types.